pandas/core/arrays/datetimelike.py

```diff
@@ -745,6 +745,56 @@ def _from_factorized(cls, values, original):
     def _values_for_argsort(self):
         return self._data
 
+    def shift(self, periods=1, fill_value=None, axis=0):
+        if not self.size or periods == 0:
```

Comment:
can you type things; I think we have an inherited doc-string?

Reply:
adding the docstring is easy, typing i tried and ended up punting on